Sources/AST/Type.swift

```diff
@@ -170,6 +181,13 @@ public struct Type: ASTNode {
     sourceLocation = identifier.sourceLocation
   }
 
+  public static func selfType(token: Token) -> Type {
```

Comment:
Why is this a static function rather than a constructor?

Reply:
Hmm. We could use the Type(rawType, identifier) constructor inlined, without the need of selfType at all, but thought it would be more descriptive to have it constructed like this. I can change it, since it's not really consistent with the rest of the codebase.